cmd/secretless-broker/main.go

```diff
@@ -41,5 +77,68 @@ func main() {
 	// Either the flag or the arg should be enough to show the version
 	params.ShowVersion = *showVersion || flag.Arg(0) == "version"
 
-	secretless.Start(&params, plugin.GetManager())
+	return params
+}
+
+func NewKillSignalChan() chan os.Signal {
+	signalChannel := make(chan os.Signal, 1)
+	signal.Notify(signalChannel,
+		syscall.SIGABRT,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGQUIT,
+		syscall.SIGTERM,
+	)
+	return signalChannel
+}
+
+type Stopper interface {
+	Stop()
+}
```

Comment:
Stopper,  NewKillSignalChan,  and StopOnKillSignal should be in a separate package